mindmeld/components/entity_resolver.py

```diff
@@ -562,38 +805,876 @@ def _predict_exact_match(self, entity):
+                    logger.warning(msg)
+
+        if self._use_sbert_model:
+            results = self.sbert_model.encode(phrases,
+                                              batch_size=self.batch_size,
+                                              convert_to_numpy=convert_to_numpy,
+                                              show_progress_bar=show_progress)
+
+        return results
+
+    def _encode_custom(self, sentences,
+                       batch_size: int = 32,
+                       show_progress_bar: bool = None,
+                       output_value: str = 'sentence_embedding',
+                       convert_to_numpy: bool = True,
+                       convert_to_tensor: bool = False,
+                       device: str = None,
+                       concat_last_n_layers: int = 1,
+                       normalize_token_embs: bool = False):
+        """
+        Computes sentence embeddings (Note: Method largely derived from Sentence Transformers
+            library to improve flexibility in encoding and pooling. Notably, `is_pretokenized` and
+            `num_workers` are ignored due to deprecation in their library, retrieved 23-Feb-2021)
+
+        Args:
+            sentences (list[str]): the sentences to embed
+            batch_size (int): the batch size used for the computation
+            show_progress_bar (bool): Output a progress bar when encode sentences
+            output_value (str): Default sentence_embedding, to get sentence embeddings.
+                Can be set to token_embeddings to get wordpiece token embeddings.
+            convert_to_numpy (bool): If true, the output is a list of numpy vectors. Else, it is a
+                list of pytorch tensors.
+            convert_to_tensor (bool): If true, you get one large tensor as return. Overwrites any
+                setting from convert_to_numpy
+            device: Which torch.device to use for the computation
+            concat_last_n_layers (int): number of hidden outputs to concat starting from last layer
+            normalize_token_embs (bool): if the (sub-)token embs are to be individually normalized
+
+        Returns:
+            (Union[List[Tensor], ndarray, Tensor]): By default, a list of tensors is returned.
+                If convert_to_tensor, a stacked tensor is returned. If convert_to_numpy, a numpy
+                matrix is returned.
+        """
+
+        if concat_last_n_layers != 1:
+            assert 1 <= concat_last_n_layers <= self._num_layers(self.transformer_model.auto_model)
+
+        self.transformer_model.eval()
+        if show_progress_bar is None:
+            show_progress_bar = (
+                logger.getEffectiveLevel() == logging.INFO or
+                logger.getEffectiveLevel() == logging.DEBUG
+            )
+
+        if convert_to_tensor:
+            convert_to_numpy = False
+
+        input_is_string = isinstance(sentences, str)
```

Comment:
Seems redundant since `_encode()` will always pass in a list?

Reply:
it was not necessary to convert string to list of string in `_encode()` method. So removed that. Thanks for pointing it out.